src/services/kratos/identity.ts

```diff
@@ -21,34 +23,78 @@ export const getNextPage = (link: string): number | undefined => {
 const perPage = isDev ? 3 : 50
 
-export const listIdentities = async (): Promise<IdentityPhone[] | KratosError> => {
-  try {
-    const identities: Identity[] = []
-    let totalCount = 0
+export const IdentityRepository = (): IIdentityRepository => {
+  const getIdentity = async (
+    kratosUserId: UserId,
+  ): Promise<IdentityPhone | KratosError> => {
+    let data: Identity
+
+    try {
+      const res = await kratosAdmin.adminGetIdentity(kratosUserId)
+      data = res.data
+    } catch (err) {
+      return new UnknownKratosError(err)
+    }
+
+    return toDomainIdentityPhone(data)
+  }
+
+  const listIdentities = async (): Promise<IdentityPhone[] | KratosError> => {
+    try {
+      const identities: Identity[] = []
+      let totalCount = 0
+
+      let hasNext = true
 
-    let hasNext = true
+      let page: number | undefined = 0
 
-    let page: number | undefined = 0
+      while (hasNext) {
+        // Note: this call is paginated, return 250 records `perPage` by default
+        const res = await kratosAdmin.adminListIdentities(perPage, page)
+        identities.push(...res.data)
 
-    while (hasNext) {
-      // Note: this call is paginated, return 250 records `perPage` by default
-      const res = await kratosAdmin.adminListIdentities(perPage, page)
-      identities.push(...res.data)
+        totalCount = Number(res.headers["x-total-count"])
 
-      totalCount = Number(res.headers["x-total-count"])
+        page = getNextPage(res.headers.link)
+        hasNext = page !== undefined && totalCount !== 0
+      }
 
-      page = getNextPage(res.headers.link)
-      hasNext = page !== undefined && totalCount !== 0
+      // FIXME(nb) function above return duplicated query for the first 2 calls, so removing them here
+      const uniqueIdentities = identities.filter(
+        (value, index, self) => index === self.findIndex((t) => t.id === value.id),
+      )
+      assert(totalCount == uniqueIdentities.length)
+
+      return uniqueIdentities.map(toDomainIdentityPhone)
+    } catch (err) {
+      return new UnknownKratosError(err)
+    }
+  }
+
+  // only use for non public endpoint for now
+  // because there is no index/go through all records
+  const slowFindByPhone = async (
+    phone: PhoneNumber,
+  ): Promise<IdentityPhone | KratosError> => {
+    let identities: Identity[]
+
+    try {
+      const res = await kratosAdmin.adminListIdentities()
+      identities = res.data
+    } catch (err) {
+      return new UnknownKratosError(err)
     }
 
-    // FIXME(nb) function above return duplicated query for the first 2 calls, so removing them here
-    const uniqueIdentities = identities.filter(
-      (value, index, self) => index === self.findIndex((t) => t.id === value.id),
-    )
-    assert(totalCount == uniqueIdentities.length)
+    const identity = identities.find((identity) => identity.traits.phone === phone)
+
+    if (!identity) return new PhoneIdentityInexistentError(phone)
```

Comment:
Optional nit, `PhoneIdentityNonExistentError` or `PhoneIdentityDoesNotExistError` might be better here